pymc/model_graph.py

```diff
@@ -302,164 +348,215 @@ def get_plates(self, var_names: Iterable[VarName] | None = None) -> dict[str, se
         """
         plates = defaultdict(set)
 
-        # TODO: Evaluate all RV shapes and dim_length at once.
-        #       This should help to find discrepancies, and
-        #       avoids unnecessary function compiles for deetermining labels.
+        # TODO: Evaluate all RV shapes at once
+        #       This should help find discrepencies, and
+        #       avoids unnecessary function compiles for determining labels.
+        dim_lengths: dict[str, int] = {
+            name: fast_eval(value).item() for name, value in self.model.dim_lengths.items()
+        }
 
         for var_name in self.vars_to_plot(var_names):
             v = self.model[var_name]
-            shape: Sequence[int] = fast_eval(v.shape)
-            dim_labels = []
+            shape: tuple[int, ...] = tuple(fast_eval(v.shape))
             if var_name in self.model.named_vars_to_dims:
                 # The RV is associated with `dims` information.
+                names = []
+                sizes = []
                 for d, dname in enumerate(self.model.named_vars_to_dims[var_name]):
-                    if dname is None:
-                        # Unnamed dimension in a `dims` tuple!
-                        dlen = shape[d]
-                        dname = f"{var_name}_dim{d}"
-                    else:
-                        dlen = fast_eval(self.model.dim_lengths[dname])
-                    dim_labels.append(f"{dname} ({dlen})")
-                plate_label = " x ".join(dim_labels)
+                    names.append(dname)
+                    sizes.append(dim_lengths.get(dname, shape[d]))
+
+                plate_meta = PlateMeta(
+                    names=tuple(names),
+                    sizes=tuple(sizes),
+                )
```

Comment:
I don't understand this tbh. Are we creating one plate per variable? But a plate can contain multiple variables?

Also names is ambiguous, it is dim_names? We should name it like that to distinguish from var_names?

Reply:
Ah you made plates hashable... so you mutate the same thing...

Reply:
Just working with what was there. The historical { str: set[VarName] } is created with loop which I changed to { PlateMeta : set[NodeMeta] } 
But switched to list[Plate] ultimately. 
Ideally, there could be more straight-foward path to list[Plate]

Reply:
This logic feels rather convoluted to be honest. Maybe we can take a step back and see what is actually needed. 
Step1: Collect the dim lengths and sizes of every variable we want to plot. This seems simple enough, and we can do in a loop
Step2: Merge variables that have identical dim lengths and sizes into "plates".

Would the code be more readable if we didn't try to do both things at once?

Reply:
Edit: Updated comment above

Reply:
defaultdict with dims_names and dim_lengths is same as what is currently happening. But there is a wrapper class around it. Personally, I find the class helpful and more user friendly. But I could be wrong

For instance, 
```python
Plate(
    DimInfo(names=("obs", "covariate"), sizes=(10, 5)), 
    variables=[
        NodeInfo(X, node_type=DATA), 
        NodeInfo(X_transform, node_type=DETERMINISTIC), 
        NodeInfo(tvp, node_type=FREE_RV),
    ]
)
```
over 
```python
(("obs", "covariate"), (10, 5), (X, X_transform, tvp), (DATA, DETERMINSTIC, FREE_RV))
```
lines up a bit better in my mind that the first two are related objects and the last two are related objects as well

Reply:
Sure, just thinking of how easy we make it for users to define their custom stuff. Either way seems manageable